feathr_project/feathr/_databricks_submission.py

```diff
@@ -192,4 +198,43 @@ def get_job_result_uri(self) -> str:
                                headers=self.auth_headers, params={'run_id': str(self.res_job_id)})
         custom_tags = result.json()['cluster_spec']['new_cluster']['custom_tags']
         assert custom_tags is not None
-        return custom_tags[OUTPUT_PATH_TAG]
\ No newline at end of file
+        return custom_tags[OUTPUT_PATH_TAG]
+
+    def download_result(self, result_path: str, local_folder: str):
+        """
+        Supports downloading files from the result folder. Only support paths starts with `dbfs:/` and only support downloading files in one folder (per Spark's design, everything will be in the result folder in a flat manner)
+        """
+        if not result_path.startswith('dfbs'):
+            RuntimeError('Currently only paths starting with dbfs is supported for downloading results from a databricks cluster. The path should start with \"dbfs:\" .')
+            # listing all the files in a folder: https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/latest/dbfs#--list
+            result = requests.get(url=self.workspace_instance_url+'/api/2.0/dbfs/list',
+                              headers=self.auth_headers,  params={ 'path': result_path})
+            dbfs_files = result.json()['files']
+            for file_path in dbfs_files:
+                # each file_path would be a dict of this type: https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/latest/dbfs#dbfsfileinfo
+                dbfs_file_path, dbfs_file_size, local_file_path = file_path['path'], file_path['file_size'], os.path.join(local_folder, os.path.basename(file_path['path']))
+                with open(local_file_path, 'wb') as file_obj:
+                    downloaded_size = 0
+                    # Loop until we've downloaded the whole file
+                    while downloaded_size < dbfs_file_size:
+                        chunk = self._read_single_chunk(path=dbfs_file_path, offset=downloaded_size, length=DATABRICKS_MB_BYTES)
+                        file_obj.write(base64.b64decode(chunk.data))
+                        downloaded_size += chunk.bytes_read
```

Comment:
what if the file is too large? Can we notify how much data are we going to download and notify the user the progres?